config/configgrpc/configgrpc.go

```diff
@@ -83,7 +66,7 @@ type GRPCClientSettings struct {
 	Endpoint string `mapstructure:"endpoint"`
 
 	// The compression key for supported compression types within collector.
-	Compression string `mapstructure:"compression"`
+	Compression middleware.CompressionType `mapstructure:"compression"`
```

Comment:
There is a problem with this, similar in http case. We are exposing publicly an "internal" type (under internal/** path). We need to move `middleware.CompressionType` in a public package. We can do this in a separate PR, but I want you to have a tracking issue.

Reply:
Posted a new issue #4627